pkg/api/api.proto

```diff
@@ -513,3 +513,20 @@ message OptionalBool {
 message OptionalFileMode {
   uint32 value = 1;
 }
+
+// CompoundFieldOwners tracks 'plugin ownership' of compound fields, typically
+// maps or slices.
+message CompoundFieldOwners {
+    map<string, string> owners = 1;
```

Comment:
Could you help clarify what the expected keys and values would be?

Reply:
I'll add clarifying comments.

Reply:
Added comments explaining what the map keys and values are.